### PR TITLE
Added SoftDeleteRetentionInDays, CustomizedRecoverable and recoverabl…

### PR DESCRIPTION
**Pull request by gdlxm**
…eDays

Added SoftDeleteRetentionInDays, CustomizedRecoverable and recoverableDays

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14-preview/keyvault.json [hunk 1]
```diff
@@ -877,6 +877,11 @@
           "type": "boolean",
           "description": "Property to specify whether the 'soft delete' functionality is enabled for this key vault. It does not accept false value."
         },
+        "softDeleteRetentionInDays": {
```
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2018-02-14-preview/keyvault.json]: Thank you fixing the data plane, but this management plane already shipped as well. I see a stable/2018-02-14, which means you shouldn't be modifying a preview for this version either. /cc @johanste 

OpenAPI Hub can be used to create new versions for the management plane as well, like I did for the data plane.

**Review by heaths — CHANGES_REQUESTED** [review]
The management plane needs to have a new revision as well. I can show you how, or start one you can target. Mainly, just create a new directory like preview/2020-01-23, copy stable/2019-09-01 to it, and make your changes to it.

**Review by heaths — APPROVED** [review]
I diff'd the stable/2019-09-01 and preview/2020-01-23 directories and they look good. I see only version changes and the softDeleteInDays changes.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2020-01-23/keyvault.json [hunk 2]
```diff
@@ -0,0 +1,1370 @@
...
+          "type": "boolean",
+          "description": "Property to specify whether Azure Disk Encryption is permitted to retrieve secrets from the vault and unwrap keys."
+        },
+        "enabledForTemplateDeployment": {
+          "type": "boolean",
+          "description": "Property to specify whether Azure Resource Manager is permitted to retrieve secrets from the key vault."
+        },
+        "enableSoftDelete": {
+          "type": "boolean",
+          "description": "Property to specify whether the 'soft delete' functionality is enabled for this key vault. If it's not set to any value(true or false) when creating new key vault, it will be set to true by default. Once it's been set to true value, it can NOT be reverted to false."
```
Comment (johanste) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2020-01-23/keyvault.json]: Please specify the `default` value as True. It implies exactly what you are explaining in the description.

**Review by schaabs — APPROVED** [review]

**Comment by allenjzhang** [comment]
Manual merge into feature branch until we fix the permission issue.

**Pull request merged**